Navigating to user profile via link does nothing

## Error in Dank

**java.lang.RuntimeException** in **UrlRouter.java:64**
Unable to start activity ComponentInfo{me.thanel.dank/me.saket.dank.deeplinks.DeepLinkHandlingActivity}: java.lang.UnsupportedOperationException: Use forLink(RedditUserLink) instead.

[View on Bugsnag](https://app.bugsnag.com/tunous/dank/errors/5d744a688ab3be001a23a6f2?event_id=5d744a68004f3868759e0000&i=gh&m=ci)

## Stacktrace

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{me.thanel.dank/me.saket.dank.deeplinks.DeepLinkHandlingActivity}: java.lang.UnsupportedOperationException: Use forLink(RedditUserLink) instead.
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2951)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3086)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6719)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:495)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: java.lang.UnsupportedOperationException: Use forLink(RedditUserLink) instead.
        at me.saket.dank.ui.UrlRouter.forLink(UrlRouter.java:64)
        at me.saket.dank.deeplinks.DeepLinkHandlingActivity.onCreate(DeepLinkHandlingActivity.java:61)
        at android.app.Activity.performCreate(Activity.java:7144)
        at android.app.Activity.performCreate(Activity.java:7135)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1272)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2931)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3086)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6719)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:495)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

[View full stacktrace](https://app.bugsnag.com/tunous/dank/errors/5d744a688ab3be001a23a6f2?event_id=5d744a68004f3868759e0000&i=gh&m=ci)

*Created automatically via Bugsnag*

--- a/app/src/main/java/me/saket/dank/deeplinks/DeepLinkHandlingActivity.java
+++ b/app/src/main/java/me/saket/dank/deeplinks/DeepLinkHandlingActivity.java
@@ -7,6 +7,7 @@ import android.content.pm.ShortcutManager;
 import android.net.Uri;
 import android.os.Build;
 import android.os.Bundle;
+import android.widget.Toast;
 
 import com.airbnb.deeplinkdispatch.DeepLinkHandler;
 import com.airbnb.deeplinkdispatch.DeepLinkResult;
@@ -18,9 +19,11 @@ import me.saket.dank.di.Dank;
 import me.saket.dank.ui.DankActivity;
 import me.saket.dank.ui.UrlRouter;
 import me.saket.dank.ui.appshortcuts.AppShortcut;
+import me.saket.dank.ui.preferences.DefaultWebBrowser;
 import me.saket.dank.ui.subreddit.SubredditActivity;
 import me.saket.dank.urlparser.Link;
 import me.saket.dank.urlparser.RedditSubredditLink;
+import me.saket.dank.urlparser.RedditUserLink;
 import me.saket.dank.urlparser.UrlParser;
 
 @DeepLinkHandler(AppDeepLinkModule.class)
@@ -58,9 +61,14 @@ public class DeepLinkHandlingActivity extends DankActivity {
         Uri uri = getIntent().getData();
         if (uri != null) {
           Link parsedLink = urlParser.get().parse(uri.toString());
-          urlRouter.get().forLink(parsedLink)
-              .expandFromBelowToolbar()
-              .open(this);
+          if (parsedLink instanceof RedditUserLink) {
+            Toast.makeText(this, "Viewing user profiles is currently unsupported", Toast.LENGTH_LONG).show();
+            startActivity(DefaultWebBrowser.DANK_INTERNAL_BROWSER.intentForUrl(this, parsedLink.unparsedUrl(), null));
+          } else {
+            urlRouter.get().forLink(parsedLink)
+                .expandFromBelowToolbar()
+                .open(this);
+          }
         }
       }
     }

--- a/app/src/main/java/me/saket/dank/ui/preferences/DefaultWebBrowser.java
+++ b/app/src/main/java/me/saket/dank/ui/preferences/DefaultWebBrowser.java
@@ -4,6 +4,7 @@ import android.content.Context;
 import android.content.Intent;
 import android.graphics.Rect;
 
+import androidx.annotation.Nullable;
 import androidx.annotation.StringRes;
 
 import me.saket.dank.R;
@@ -14,19 +15,19 @@ import me.saket.dank.utils.Intents;
 public enum DefaultWebBrowser {
   DANK_INTERNAL_BROWSER(R.string.userprefs_externallinks_internal_browser) {
     @Override
-    public Intent intentForUrl(Context context, String url, Rect expandFromRect) {
+    public Intent intentForUrl(Context context, String url, @Nullable Rect expandFromRect) {
       return WebViewActivity.intent(context, url, expandFromRect);
     }
   },
   CHROME_CUSTOM_TABS(R.string.userprefs_externallinks_chrome_custom_tabs) {
     @Override
-    public Intent intentForUrl(Context context, String url, Rect expandFromRect) {
+    public Intent intentForUrl(Context context, String url, @Nullable Rect expandFromRect) {
       return ChromeCustomTabsForwardingActivity.intent(context, url);
     }
   },
   DEVICE_DEFAULT(R.string.userprefs_externallinks_default_web_browser) {
     @Override
-    public Intent intentForUrl(Context context, String url, Rect expandFromRect) {
+    public Intent intentForUrl(Context context, String url, @Nullable Rect expandFromRect) {
       return Intents.createForOpeningUrl(url);
     }
   };
@@ -38,7 +39,7 @@ public enum DefaultWebBrowser {
     this.displayName = displayName;
   }
 
-  public Intent intentForUrl(Context context, String url, Rect expandFromRect) {
+  public Intent intentForUrl(Context context, String url, @Nullable Rect expandFromRect) {
     throw new AbstractMethodError();
   }
 }

--- a/app/src/main/java/me/saket/dank/ui/webview/WebViewActivity.java
+++ b/app/src/main/java/me/saket/dank/ui/webview/WebViewActivity.java
@@ -36,7 +36,7 @@ public class WebViewActivity extends DankPullCollapsibleActivity {
   @BindView(R.id.webviewfallback_webview) WebView webView;
   @BindView(R.id.webviewfallback_progress) ProgressBar progressView;
 
-  public static Intent intent(Context context, String url, Rect expandFromShape) {
+  public static Intent intent(Context context, String url, @Nullable Rect expandFromShape) {
     Intent intent = new Intent(context, WebViewActivity.class);
     intent.putExtra(KEY_URL, url);
     intent.putExtra(KEY_EXPAND_FROM_SHAPE, expandFromShape);

